e2e/internal/keyexec/keyexec.go

```diff
@@ -172,118 +177,127 @@ func ExportPrivateKey(t *testing.T, kpath, num string, armor bool) (string, []by
 	}
 
 	c.Send(num)
-	c.Send("e2etests\n")
+	c.Send(E2ePrivatePass + "\n")
 
 	err = cmd.Wait()
-	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+	cm := fmt.Sprintf("%s\n%s", testenv.CmdPath, strings.Join(exportCmd, " "))
 
 	return cm, outErr.Bytes(), err
 }
 
 // RunKeyCmd will run a 'singularty key' command, with any args that are set in commands.
-func RunKeyCmd(t *testing.T, cmdPath string, commands []string, file, stdin string) (string, []byte, error) {
+func RunKeyCmd(t *testing.T, commands []string, stdin string) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
 	argv := []string{"key"}
 	argv = append(argv, commands...)
 
-	if file != "" {
-		argv = append(argv, file)
-	}
-
-	cmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(argv, " "))
-	execKey := exec.Command(cmdPath, argv...)
+	cm := fmt.Sprintf("%s\n%s", testenv.CmdPath, strings.Join(argv, " "))
+	execKey := exec.Command(testenv.CmdPath, argv...)
 
-	stdinRun, err := ioutil.TempFile("", "")
+	c, err := expect.NewConsole()
 	if err != nil {
-		t.Fatalf("%v", err)
+		t.Fatal("unable to start new console: ", err)
 	}
-	defer stdinRun.Close()
+	defer c.Close()
 
-	_, err = io.WriteString(stdinRun, stdin)
-	if err != nil {
-		t.Fatalf("%v", err)
-	}
+	outErr := bytes.NewBuffer(nil)
+
+	execKey.Stdin = c.Tty()
+
+	execKey.Stderr = outErr
+	execKey.Stdout = outErr
+
+	go func() {
+		c.ExpectEOF()
+	}()
 
-	_, err = stdinRun.Seek(0, os.SEEK_SET)
+	err = execKey.Start()
 	if err != nil {
-		t.Fatalf("%v", err)
+		t.Fatalf("unable to run command: %v", err)
 	}
 
-	execKey.Stdin = stdinRun
-	out, err := execKey.CombinedOutput()
+	c.Send(stdin)
+
+	err = execKey.Wait()
 
-	return cmd, out, err
+	return cm, outErr.Bytes(), err
 }
 
 // QuickTestExportImportKey will export a private, and public key (0), and then import them. This is used after
-// generating a newpair.
-func QuickTestExportImportKey(t *testing.T) {
+// generating a newpair. keyNum is the key number to test. Its a string, so the number must be used in "", and
+// end with a '\n', eg. "1\n" will test key 1.
```

Comment:
it would be better to make keyNum an int, and have this function convert it to a string and add the necessary newline.